src/action/common/provision_determinate_nixd.rs

```diff
@@ -4,20 +4,21 @@ use std::path::PathBuf;
 use tokio::fs::{create_dir_all, remove_file};
 use tracing::{span, Span};
 
-use crate::action::common::configure_determinate_nixd_init_service::DETERMINATE_NIXD_SERVICE_SRC;
 use crate::action::{
     Action, ActionDescription, ActionError, ActionErrorKind, ActionTag, StatefulAction,
 };
 
+#[cfg(target_os = "linux")]
 const DETERMINATE_NIXD_BINARY_PATH: &str = "/nix/determinate/determinate-nixd";
+#[cfg(target_os = "macos")]
+const DETERMINATE_NIXD_BINARY_PATH: &str = "/usr/local/bin/determinate-nixd";
```

Comment:
nit: you've previously complained about how cfg-ing things makes it harder to develop with -- could we not cfg these and instead match on platform like we do in other places?